controllers/userController.js

```diff
@@ -0,0 +1,106 @@
+const User = require('../models/userModel')
+const generateToken = require('../utils/generateTokens');
+
+//@desc   signup new user instructor
+//@route  POST /api/register/instructor
+//@access Public
+const signupInstructor = async (req, res) => {
+    const { firstName, lastName, dob, email, password } = req.body;
+
+    const userExist = await User.findOne({ email });
+    if (userExist) {
+        res.status(400);
+        throw new Error('User already exists');
+    }
+
+    const instructor = await User.create({
+        firstName,
+        lastName,
+        dob,
+        email,
+        password,
+        type: 'instructor',
```

Comment:
I recommend using an object approach to define user roles. Instead of using string literals like 'instructor', consider using constants or enums within an object like :

```
const Role = {
    Instructor: 'instructor',
    Student: 'student',
    Admin: 'admin'
};
```
With this approach, you can easily reference user roles throughout your code using the constants defined in the Role object, such as:
```
Role.Instructor 
Role.Student. 
```
This reduces the likelihood of errors and enhances the security by preventing cases like click jacking.